app/questionnaire/questionnaire_store_updater.py

```diff
@@ -550,3 +488,209 @@ def _get_chronological_section_dependents(self) -> list:
         return sorted(
             self.dependent_sections, key=lambda x: sections.index(x.section_id)
         )
+
+    def set_supplementary_data(self, to_set: MutableMapping) -> None:
+        """
+        Used to set or update the supplementary data whenever the sds endpoint is called
+        (Which should be once per session, but only if the sds_dataset_id has changed)
+
+        this updates ListStore to add/update any lists for supplementary data and stores the
+        identifier -> list_item_id mappings in the supplementary data store to use in the payload at the end
+        """
+        list_mappings: dict[str, list[SupplementaryDataListMapping]] = {}
+        modified_lists: set[str] = set()
+
+        if self._questionnaire_store.data_stores.supplementary_data_store.list_mappings:
```

Comment:
Given the other stores all have an attribute assigned, we should do the same for sds for consistency.

```py
self._answer_store = self._questionnaire_store.data_stores.answer_store
self._list_store = self._questionnaire_store.data_stores.list_store
self._progress_store = self._questionnaire_store.data_stores.progress_store
```

But given you are updating the literal instance, you will need to use `@property` to ensure there are no stale references. It can also be private like the others.

```py
@property
def _supplementary_data_store(self) -> SupplementaryDataStore:
    return self._questionnaire_store.data_stores.supplementary_data_store
```